nexus_constructor/instrument_view.py

```diff
@@ -23,6 +23,12 @@ class InstrumentView(QWidget):
                    argument in order to appease Qt Designer.
     """
 
+    def __del__(self):
```

Comment:
It is my understanding that Python does not give any guarantees about `__del__` being called. (Do you have any wise words here @matthew-d-jones ?)

On the other hand, if it works, why not use the solution that we have?

Reply:
Yes `__del__` is not the right place to do this, it only gets called when/if the garbage collector deals with the object.

Reply:
Calling this code in the application exit function would be better.